### PR TITLE
WIP: resolve "Deep learning for symbolic mathematics"

### PR DESCRIPTION
**Pull request by zzj0402**
- [ ] implementing julia equivalent for https://github.com/facebookresearch/SymbolicMathematics/blob/master/src/model/transformer.py

### CODE REVIEW

File: src/symbolic.jl [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  "vocab_size"                   => 30522,
+  "num_attention_heads"          => 12,
+  "hidden_dropout_prob"          => 0.1,
+  "intermediate_size"            => 3072,
+)
+
+# create bert model like pretrain struct
+# you can define the bert model in the way you like and wrap it with TransformerModel
+function create_bert()
+    global config
```
Comment (ChrisRackauckas) [inline src/symbolic.jl:19]: I don't get this config dictionary. Why not just have these be keyword arguments to `create_bert`?

Comment (zzj0402) [reply src/symbolic.jl:19]: Hi Chris, this is just a starter boilerplate from Internet. I will mark ready for review when you can feedback.

Comment (ChrisRackauckas) [reply src/symbolic.jl:19]: oh okay, no worries.

**Comment by zzj0402** [comment]
```
We use a transformer model (Vaswani et al., 2017) with 8 attention heads, 6 layers, and a dimensionality of
512.
```
https://arxiv.org/pdf/1912.01412.pdf

**Comment by zzj0402** [comment]
## Input Tokenization
## Encoder
## Decoder

**Comment by zzj0402** [comment]
```
julia> using Transformers
[ Info: Precompiling Transformers [21ca0261-441d-5938-ace7-c90938fde4d4]
ERROR: LoadError: LoadError: UndefVarError: Tracker not defined
Stacktrace:
 [1] include(::Module, ::String) at .\Base.jl:377
 [2] include(::String) at C:\Users\zzj04\.julia\packages\Transformers\2BZJs\src\Transformers.jl:1
 [3] top-level scope at C:\Users\zzj04\.julia\packages\Transformers\2BZJs\src\Transformers.jl:33
 [4] include(::Module, ::String) at .\Base.jl:377
 [5] top-level scope at none:2
 [6] eval at .\boot.jl:331 [inlined]
 [7] eval(::Expr) at .\client.jl:449
 [8] top-level scope at .\none:3
in expression starting at C:\Users\zzj04\.julia\packages\Transformers\2BZJs\src\fix\batchedmul.jl:3
in expression starting at C:\Users\zzj04\.julia\packages\Transformers\2BZJs\src\Transformers.jl:33
ERROR: Failed to precompile Transformers [21ca0261-441d-5938-ace7-c90938fde4d4] to C:\Users\zzj04\.julia\compiled\v1.4\Transformers\gPQmm_KBlm6.ji.
Stacktrace:
 [1] error(::String) at .\error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at .\loading.jl:1272
 [3] _require(::Base.PkgId) at .\loading.jl:1029
 [4] require(::Base.PkgId) at .\loading.jl:927
 [5] require(::Module, ::Symbol) at .\loading.jl:922
```

**Comment by ChrisRackauckas** [comment]
What version do you have?

**Comment by zzj0402** [comment]
> What version do you have?

Julia Version 1.4.2 (2020-05-23)
Transformers v0.1.3

**Comment by ChrisRackauckas** [comment]
Is this in a restarted REPL? That version doesn't reference Tracker. Also check your Flux version

**Comment by zzj0402** [comment]
> Is this in a restarted REPL? That version doesn't reference Tracker. Also check your Flux version

Flux is working fine.  What do you mean restarted REPL? What is Tracker?

**Comment by zzj0402** [comment]
Dependency issue on Flux
https://stackoverflow.com/questions/59765831/julia-flux-issue-undefvarerror-tracker-not-defined

**Pull request closed without merging**